### PR TITLE
add uppercase name JSON field test

### PR DESCRIPTION
**Pull request by frogcjn**
add uppercase name JSON field test

### CODE REVIEW

**Comment by frogcjn** [comment]
https://github.com/sequelize/sequelize/issues/6406

File: test/integration/dialects/postgres/dao.test.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ if (dialect.match(/^postgres/)) {
         document: { type: DataTypes.HSTORE, defaultValue: { default: "'value'" } },
         phones: DataTypes.ARRAY(DataTypes.HSTORE),
         emergency_contact: DataTypes.JSON,
+        emergencyCntact: DataTypes.JSON,
```
Comment (janmeier) [inline test/integration/dialects/postgres/dao.test.js]: typo

File: test/integration/dialects/postgres/dao.test.js [hunk 2]
```diff
@@ -251,6 +252,20 @@ if (dialect.match(/^postgres/)) {
             expect(user.emergency_contact.name).to.equal('joe');
           });
       });
+      
+      it('should be able to query using dot syntax with uppercase name', function() {
+        var self = this;
+
+        return this.sequelize.Promise.all([
```
Comment (janmeier) [inline test/integration/dialects/postgres/dao.test.js]: Could you add a test for attributes as well

Comment (frogcjn) [inline test/integration/dialects/postgres/dao.test.js]: like findAll( { attributes: { } } in my issue?

Comment (janmeier) [inline test/integration/dialects/postgres/dao.test.js]: Yep

**Comment by janmeier** [comment]
A unit test would also be nice :) https://github.com/sequelize/sequelize/tree/master/test/unit/sql

File: test/integration/utils.test.js [hunk 3]
```diff
@@ -194,13 +194,13 @@ describe(Support.getTestDialectTeaser('Utils'), function() {
...
         };
-        var expected = "metadata#>>'{language}' = 'icelandic' and metadata#>>'{pg_rating,dk}' = 'G' and another_json_field#>>'{x}' = '1'";
+        var expected = "\"metadata\"#>>'{language}' = 'icelandic' and metadata#>>'{pg_rating,dk}' = 'G' and another_json_field#>>'{x}' = '1'";
         expect(queryGenerator.handleSequelizeMethod(new Utils.Json(conditions))).to.deep.equal(expected);
       });
 
       it('successfully parses a string using dot notation', function() {
         var path = 'metadata.pg_rating.dk';
-        expect(queryGenerator.handleSequelizeMethod(new Utils.Json(path))).to.equal("metadata#>>'{pg_rating,dk}'");
+        expect(queryGenerator.handleSequelizeMethod(new Utils.Json(path))).to.equal("\"metadata\"#>>'{pg_rating,dk}'");
```
Comment (mickhansen) [inline test/integration/utils.test.js]: Use template strings to avoid having to escape both `"` and `'`

Comment (frogcjn) [inline test/integration/utils.test.js]: `${}` this one?

**Comment by frogcjn** [comment]
@janmeier since the unit test of postgresql is not including json related request, I'll leave it.

**Comment by frogcjn** [comment]
@janmeier , @mickhansen , all test passed! Including new test!

**Comment by janmeier** [comment]
> @janmeier since the unit test of postgresql is not including json related test, I'll leave it.

Alright, we do have some sql generation tests in utils test it seems, so it should be fine

**Comment by janmeier** [comment]
If you add a changelog entry, this should be good to go

**Comment by frogcjn** [comment]
@janmeier changeling added

File: changelog.md [hunk 4]
```diff
@@ -33,6 +33,8 @@
 - [SECURITY] `GEOMETRY` and `GEOGRAPHY` SQL injection attacks [#6194](https://github.com/sequelize/sequelize/issues/6194)
 - [FIXED] `DECIMAL` now supports `UNSIGNED` / `ZEROFILL` (MySQL) [#2038](https://github.com/sequelize/sequelize/issues/2038)
 - [FIXED] Generate correct SQL of nested include when quoteIdentifiers is false. (Postgres) [#6351](https://github.com/sequelize/sequelize/issues/6351)
+- [FIXED] Generate correct SQL of JSON attributes with quote.
```
Comment (sushantdhiman) [inline changelog.md]: **Please change to :** Generate correct SQL for JSON attributes with quote

**Pull request merged**